src/mon/OSDMonitor.cc

```diff
@@ -3100,7 +3102,7 @@ void OSDMonitor::do_application_enable(int64_t pool_id,
     p = pending_inc.new_pools[pool_id];
   }
 
-  p.application_metadata.insert({app_name, {}});
+  p.application_metadata.insert({app_name, {{app_key, app_value}}});
```

Comment:
Doesn’t this add empty entries in the default case? I think we want to only add the key/value conditionally if they have length.